ui/src/app/applications/components/utils.tsx

```diff
@@ -894,3 +904,15 @@ export const BASE_COLORS = [
     '#4B0082', // purple
     '#964B00' // brown
 ];
+
+export function useData<T>(getData: () => Promise<T>, init?: T): T {
```

Comment:
The `DataLoader` allows rendering the "loading" indicator and error message with the retry button. Please add the same functionality to `useData`.